frodo.js

```diff
@@ -1,22 +1,23 @@
 /**
```

Comment:
SCRIPT-I like how you've used promise chaining here, it makes the code flow very clear.